File: certif-parent/certif-domain/src/main/java/com/certifapp/domain/port/output/CertificationRepository.java
// certif-parent/certif-domain/src/main/java/com/certifapp/domain/port/output/CertificationRepository.java
package com.certifapp.domain.port.output;

import com.certifapp.domain.model.certification.Certification;
import java.util.List;
import java.util.Optional;

/** Output port: persistence operations for {@link Certification}. */
public interface CertificationRepository {
    Optional<Certification> findById(String id);
    List<Certification>     findAll(boolean activeOnly);
    Certification           save(Certification certification);
}

File: certif-parent/certif-android/app/src/main/java/com/certifapp/android/domain/repository/CertificationRepository.kt
// certif-parent/certif-android/app/src/main/java/com/certifapp/android/domain/repository/CertificationRepository.kt
package com.certifapp.android.domain.repository

import com.certifapp.android.domain.model.Certification
import kotlinx.coroutines.flow.Flow

/**
 * Domain port for certification data.
 * Implemented by CertificationRepositoryImpl (data layer).
 * Uses Flow for offline-first: emits cached data first, then remote.
 */
interface CertificationRepository {
    fun getCertifications(): Flow<List<Certification>>
    suspend fun getCertificationById(id: String): Certification?
    suspend fun refreshCertifications()
}
